### PR TITLE
intelligently resize screenshot and source tree widths

### PR DESCRIPTION
**Pull request by jlipps**
this has also been bugging me. with this change there's no more horizontal or vertical space gap to the right or below the screenshot image.

### CODE REVIEW

File: app/renderer/components/Inspector/Inspector.js [hunk 1]
```diff
@@ -1,4 +1,5 @@
 import React, { Component } from 'react';
+import { debounce } from 'lodash';
```
Comment (mykola-mokhnach) [inline app/renderer/components/Inspector/Inspector.js]: I thought we usually do `import _ from 'lodash'`

Comment (jlipps) [reply app/renderer/components/Inspector/Inspector.js]: elsewhere in AD we do it this way, so i'm just following suit.

File: app/renderer/components/Inspector/Inspector.js [hunk 2]
```diff
@@ -24,6 +25,38 @@ export default class Inspector extends Component {
...
+  }
+
+  updateSourceTreeWidth () {
+    // the idea here is to keep track of the screenshot image width. if it has
+    // too much space to the right or bottom, adjust the max-width of the
+    // screenshot container so the source tree flex adjusts to always fill the
+    // remaining space. This keeps everything looking tight.
+    if (this.screenAndSourceEl) {
+      const maxScreenshotWidth = 500;
+      //const buffer = 10;
```
Comment (mykola-mokhnach) [inline app/renderer/components/Inspector/Inspector.js]: do we need this comment?

File: app/renderer/components/Inspector/Inspector.js [hunk 3]
```diff
@@ -24,6 +25,38 @@ export default class Inspector extends Component {
...
+    this.updateSourceTreeWidth = debounce(this.updateSourceTreeWidth.bind(this), 50);
+  }
+
+  updateSourceTreeWidth () {
+    // the idea here is to keep track of the screenshot image width. if it has
+    // too much space to the right or bottom, adjust the max-width of the
+    // screenshot container so the source tree flex adjusts to always fill the
+    // remaining space. This keeps everything looking tight.
+    if (this.screenAndSourceEl) {
+      const maxScreenshotWidth = 500;
```
Comment (mykola-mokhnach) [inline app/renderer/components/Inspector/Inspector.js]: this can be a global constant

File: app/renderer/components/Inspector/Inspector.js [hunk 4]
```diff
@@ -24,6 +25,38 @@ export default class Inspector extends Component {
...
+    this.lastScreenshot = null;
+    this.updateSourceTreeWidth = debounce(this.updateSourceTreeWidth.bind(this), 50);
+  }
+
+  updateSourceTreeWidth () {
+    // the idea here is to keep track of the screenshot image width. if it has
+    // too much space to the right or bottom, adjust the max-width of the
+    // screenshot container so the source tree flex adjusts to always fill the
+    // remaining space. This keeps everything looking tight.
+    if (this.screenAndSourceEl) {
```
Comment (mykola-mokhnach) [inline app/renderer/components/Inspector/Inspector.js]: how about `if (!this.screenAndSourceEl) return`. Same below

File: app/renderer/components/Inspector/Inspector.js [hunk 5]
```diff
@@ -24,6 +25,38 @@ export default class Inspector extends Component {
...
+        const imgRect = img.getBoundingClientRect();
+        const screenshotRect = screenshotBox.getBoundingClientRect();
+        screenshotBox.style.flexBasis = `${imgRect.width}px`;
+        if (imgRect.width < screenshotRect.width) {
+          screenshotBox.style.maxWidth = `${imgRect.width}px`;
+        } else if (imgRect.height < screenshotRect.height) {
+          // get what the img width would be if it fills screenshot box height
+          const attemptedWidth = (screenshotRect.height / imgRect.height) * imgRect.width;
+          if (attemptedWidth > maxScreenshotWidth) {
+            screenshotBox.style.maxWidth = `${maxScreenshotWidth}px`;
```
Comment (mykola-mokhnach) [inline app/renderer/components/Inspector/Inspector.js]: I'd rather use a ternary operator here

Comment (mykola-mokhnach) [reply app/renderer/components/Inspector/Inspector.js]:
```
screenshotBox.style.maxWidth = `${attemptedWidth > maxScreenshotWidth ? maxScreenshotWidth : attemptedWidth}px`
```

File: app/renderer/components/Inspector/Inspector.js [hunk 6]
```diff
@@ -40,6 +75,17 @@ export default class Inspector extends Component {
...
     this.props.applyClientMethod({methodName: 'source'});
     this.props.getSavedActionFramework();
+    window.addEventListener('resize', this.updateSourceTreeWidth);
+  }
+
+  componentDidUpdate () {
+    const {screenshot} = this.props;
+    // only update when the screenshot changed, not for any other kind of
+    // update
+    if (screenshot !== this.lastScreenshot) {
```
Comment (mykola-mokhnach) [inline app/renderer/components/Inspector/Inspector.js]: should we also check here if `screenshot` is defined?

Comment (jlipps) [reply app/renderer/components/Inspector/Inspector.js]: it shouldn't matter. if it is not defined, our `(!img)` check will bail us out later on.

**Pull request merged**